src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerProgressReporter.cs

```diff
@@ -39,7 +39,10 @@ public void UpdatePendingItemCount(int pendingItemCount)
                 }
             }
 
-            public IDisposable Evaluating()
+            /// <summary>
+            /// return RAII object that marking evaluation scope
+            /// </summary>
+            public IDisposable GetEvaluatingScope()
```

Comment:
How about thsi for the comment:

> Allows the solution crawler to start evaluating work enqueued to it.  Returns an IDisposable that the caller *must* dispose of to indicate that it no longer needs the crawler to continue evaluating.  Multiple callers can call into this simultaneously.  Only when the last one actually disposes the scope-object will the crawler actually revert back to the paused state where no work proceeds.

Reply:
I am not sure why this kind of detail needs to be leaked to call site. it is using ref count to make sure caller doesn't need to care all those but just wrap code with usings.

Reply:
but sure. why not.

Reply:
> I am not sure why this kind of detail needs to be leaked to call site. 

Because it woudl be a bug if the call site didn't dispose of the object.  If a method has an invariant the caller must abide by, it's usually a good thing to doc it :)  Imagine if you left the team and someone else had to come figure this out.  They wouldn't necessarily know what the rules were here and the docs make it clearer what the intent is and what's necessary for *correct* behavior.  Givne this affects state/threading/concurrency, it's def important (especially if this code gets updated in the future).